core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -127,10 +127,10 @@ object Validator {
       fragmentMap  <- validateFragments(fragments)
       selectionSets = collectSelectionSets(operations.flatMap(_.selectionSet) ++ fragments.flatMap(_.selectionSet))
       context       = Context(document, rootType, operations, fragmentMap, selectionSets, variables)
+      _            <- validateFragmentSpreads(context)
```

Comment:
We need to move this, otherwise we run this before antyhing else, otherwise we run the risk of trying to resolve cyclic fragments for variables.